.github/workflows/release.yml

```diff
@@ -0,0 +1,317 @@
+    defaults:
+      run:
+        working-directory: demos/npm
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Cache npm-demo dependencies
+        id: cache-node-modules-npm-demo
+        uses: actions/cache@v4
+        with:
+          path: demos/npm/node_modules
+          key: node-npm-demo-${{ hashFiles('demos/npm/package-lock.json') }}
+
+      - name: Install npm-demo dependencies
+        if: steps.cache-node-modules-npm-demo.outputs.cache-hit != 'true'
+        run: npm ci
+
+      - name: Build npm-demo
+        run: npm run build
+
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: dist-npm-demo
+          path: demos/npm/dist/**/*
+
+  publish-to-npm:
+    needs: [build-web, build-lib]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Download web artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: dist-web
+          path: browser/dist
+
+      - name: Download lib artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: dist-web
+          path: lib/dist
+
+        # Step will fail if the version is invalid, github.ref_name is the tag name (v.*.*.*)
+      - name: Patch version
+        run: ./scripts/patch-version.sh "${{ github.ref_name }}"
+
+      - name: Setup registry access
+        run: |
+          echo '@optable:registry=https://registry.npmjs.org/' > ~/.npmrc
+          echo "//registry.npmjs.org/:_authToken=${{ secrets.GITHUB_TOKEN }}" >> ~/.npmrc
+
+      - name: Publish to NPM
+        run: npm publish --access public
+
+  deploy-demo:
+    needs: [publish-to-npm, build-vanilla-demo, build-react-demo, build-npm-demo]
+    permissions:
+      contents: "read"
+      id-token: "write"
+    runs-on: ubuntu-latest
+    env:
+      gcp-project-id: 118585658141
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Get sdk artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: dist-web
+          path: browser/dist
+
+      - name: Get vanilla-demo artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: dist-demo-vanilla
+          path: demos
+
+      - name: Get react-demo artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: dist-react-demo
+          path: demos/react/dist
+
+      - name: Get npm-demo artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: dist-npm-demo
+          path: demos/npm/dist
+
+      - uses: "google-github-actions/auth@v2"
+        id: auth
+        with:
+          workload_identity_provider: "projects/${{ env.gcp-project-id }}/locations/global/workloadIdentityPools/optable-ci/providers/github-pool-provider"
+          service_account: "gh-ci-optable-sandbox@optable-platform-ci.iam.gserviceaccount.com"
```

Comment:
We discussed this, the standard / consistency aspect of it is still being evaluated, the reusable action uses an outdated version, it installs the gcloud which I don't need and I need the token which I can't get since it's not being exported. For those reasons we're going to keep it as is for now.